internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -452,6 +486,86 @@ func (o *Orchestrator) waitForContainerToStart(ctx context.Context, id string) e
 	}
 }
 
+func (o *Orchestrator) waitForContainerDependencies(ctx context.Context, name string, definitions map[string]ContainerDefinition) error {
+	var deps []string
+	for depName, state := range definitions[name].DependsOn {
+		deps = append(deps, fmt.Sprintf("%s->%s", depName, state))
+	}
+	logMsg := strings.Join(deps, ", ")
+	fmt.Printf("Waiting for container %q dependencies: [%s]\n", name, color.Emphasize(logMsg))
+	eg, ctx := errgroup.WithContext(ctx)
+	for name, state := range definitions[name].DependsOn {
+		name, state := name, state
+		eg.Go(func() error {
+			ctrId := o.containerID(name)
+			isEssential := definitions[name].IsEssential
+			ticker := time.NewTicker(700 * time.Millisecond)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+				case <-ctx.Done():
+					return ctx.Err()
+				}
+				switch state {
+				case ctrStateStart:
+					err := o.waitForContainerToStart(ctx, ctrId, isEssential)
+					if err != nil {
+						return err
+					}
+					if isEssential {
+						log.Successf("Successfully started container %s\n", ctrId)
+						return nil
+					}
+					fmt.Printf("non-essential container %q started successfully\n", ctrId)
+					return nil
+				case ctrStateHealthy:
+					healthy, err := o.docker.IsContainerHealthy(ctx, ctrId)
+					if err != nil {
+						if !isEssential {
+							fmt.Printf("non-essential container %q is not healthy: %v\n", ctrId, err)
```

Comment:
nit nit: just because the container could be in [non-running state](https://github.com/aws/copilot-cli/blob/9bc5f464590f5035af888e43a007fd645bc424f1/internal/pkg/docker/dockerengine/dockerengine.go#L396C2-L396C2) or maybe `docker inspect` has some unexpected error  (which does not indicate "unhealthiness" of the container):
```suggestion
							fmt.Printf("check health status for container %q: %v\n", ctrId, err)
```
The wrapped error returned from `IsContainerHealthy` is enough to tell the users whether it's because the container is indeed unhealthy, or it doesn't have health check, or other.

Reply:
also.....on a second thought, if I have a manifest like this:
```
sidecars:
  container1:
     depends_on:
         container2: healthy
     container2:
         # not essential
          healtchcheck:...
```
And then container2 turns out to be unhealthy, should I really start container1 🤔 it seems from the implementation that we will just. Curious to hear your thoughts!

Reply:
I tested this scenario. ECS tries to deploy `container2` and it becomes unhealthy. ECS does not try to start `container1`nor Circuit breaker is triggered. CloudFormation just errors our after `3 hours`.

```
Resource timed out waiting for completion
```
So `Essential-lity` and `dependsOn` are two different entities. 
Changed the logic to completely stop and remove all the containers even if `non-essential` container becomes unhealthy.